Prepare and normalize data

theolebever opened the pull request:
This Pull Request switch the program to a client / server architecture. The client read the CSV file and sends it using TLS (for security concerns) to the server, which, for now and receives it and does nothing.

theolebever committed:
First push

theolebever committed:
Update dependencies and CI accordingly

theolebever committed:
Small update on CI

theolebever committed:
Switching to server / client architecture

theolebever committed:
Adding shell script to create certificates for server and client + TLS
communication between both now

bobaiep commented on certificate.sh:
Maybe we can create an openssl.cnf

theolebever committed:
Adding checkup for integrity of chunk using SHA-256 algorithm

theolebever committed:
Update CI to perform dependencies audit

theolebever committed:
Prepare hmac data integrity verification

theolebever committed:
Loads of fixes + HMAC check for integrity working

theolebever committed:
Adding IA part of the project :
- Create clean_dataset function to:
  - Convert records to ndarray Array2<f64>
  - Split dataset into features (X) and target (y)
  - Impute missing values with column mean using impute_nan_with_mean function
  - Split dataset into train and test sets using index-based selection
- Add impute_nan_with_mean function to handle missing values

It basicaly introduces the core preprocessing steps necessary for converting the raw datasetonce received by the server to a clean format suitable for training and testing the logistic regression model.
